### ISSUE TITLE
Copycat on Google Play

### ISSUE THREAD

**Issue by Poussinou** [SimpleMobileTools/Simple-Gallery#855]
Hi @tibbi 

I found [this app named Photo Gallery & Album](https://play.google.com/store/apps/details?id=com.baniapptech.photogallery) on the Google Play Store. This guy just took your source code, added some ads in it and put it on the Store, without warning this is free software and without respecting the license...

As you are the owner of the source code, feel free to open a request to google [here](https://support.google.com/legal/contact/lr_dmca?product=googleplay) to remove this copy from the Store, if it's your wish ;) This fake developer has other free/open-source stolen apps in his list, like [this one](https://play.google.com/store/apps/details?id=com.baniapptech.jpgtopdf).

Don't forget to tell Google that:

* The app is licensed under Apachev2 license and this guy is violating it;
* He is falsely claiming he made the app;
* Provide some screenshots of your own app.

**Comment by tibbi** [comment]
thanks, reported via PlayStore app, it is easier there. You can report it too.

**Comment by Poussinou** [comment]
It's not the same thing to report it via PlayStore app than via the DMCA takedown request, but as you want 👍 

Please let me know if one day you find other copies of your apps on Google Play, so that I can track fake developers, thanks...

**Comment by tibbi** [comment]
there were some, but I wont search for them now. I added a check in the Commons library to show a fake app warning from time to time, which can obviously be removed too if they tweak the library, but it is already a lot harder as they also have to figure out why is it appearing. Copycats are expected at opensource projects, there will always be more of them.

**Comment by Poussinou** [comment]
[Here](https://play.google.com/store/apps/details?id=com.simplemobiletools.okgallery) is another one.

**Comment by Catfriend1** [comment]
Wow, that's very sad to hear. people take source code to make money and republish under a new name. The same happened to the syncthing.net project a few days ago when chinese "app maker" proxied the whole traffic to the project's community servers and rebranded it.

...

From my side: Thank you for this great app. I like it very much and feel superb using the F-Droid version! :) 👍

**Comment by tibbi** [comment]
ye, I expected it from the beginning though, so it didnt surprise me. There will always be more of them as the popularity of the app grows. Not much I can do about it.

**Comment by Poussinou** [comment]
There is also this one: https://play.google.com/store/apps/details?id=com.powerful.gallery

**Comment by tibbi** [comment]
just give them bad rating and review pointing to the original app. Thanks

**Comment by ouzowtf** [comment]
Done.

**Comment by Poussinou** [comment]
There is another one [here](https://play.google.com/store/apps/details?id=com.onetouchGallery.photogallery) and another one [here](https://play.google.com/store/apps/details?id=com.gallery.gallerynew.gallerypro.galleydownload).

**Comment by Poussinou** [comment]
Another one here: https://play.google.com/store/apps/details?id=com.ardentic.gallery

**Comment by tibbi** [comment]
I created a Reddit thread for the clones, you can find it at https://www.reddit.com/r/SimpleMobileTools/comments/csfw00/help_us_getting_rid_of_fake_clones_of_simple/ . I added some of your suggested apps, some have already been removed. Not sure if it is just some coincidence, or reporting them helped. Anyway, if you find more apps, just mention it there. Thanks

**Comment by Poussinou** [comment]
@tibbi Ok I'll post there in the future, my account is [here on Reddit](https://www.reddit.com/user/Poussinou/) for information. If you find some of them by yourself, please do not forget to post them on the discussion, so I can copy past the developer presentation page, they often have other copycats, and we can take them down easily 👍 
Thanks for your help

**Comment by Poussinou** [comment]
https://play.google.com/store/apps/details?id=com.eagle.gallery.photos.videos.album.editor.collage.maker.filter

**Comment by thgoebel** [comment]
And another:
https://play.google.com/store/apps/details?id=com.photogallery.photography.galleryapp

I reported it, but the more people, the better I guess?

**Comment by tibbi** [comment]
just give them a bad review mentioning the original app, I sadly dont think reporting will do much. Thanks though

**Comment by jaynecoding** [comment]
i am a bit surprised to read whats been said here. why are you people mad that other people have used this projects code? is this not an open source project? @tibbi why are you upset about this? i read the license under which you released this open source project, and it clearly states that anyone can use the code for commercial or for free purposes. am i wrong about this? did i miss something?

**Comment by thgoebel** [comment]
The following answer is my personal opinion from my experience with/in the open source community in general.
This copycat problem affects many open source projects, though from what I have seen mostly Android apps (though I may be biased here).

Also, I use the term “open source” in my argument below. I am aware that sometimes “free/libre software” would have been a better fit. Please don’t start a discussion over it.

There are two parts to disliking these types of copycats. The first is formal/legal, the other more informal/emotional.

### Legal reasons

Indeed, the project’s code is open source, and indeed there is a license. Github has done quite a good job at showing a summary of what any project's license means.

However, licenses do not just grant *permissions*, they do also impose *conditions* (obligations).
In the case of GPLv3, this means the modified (copycat) app must:

a) have a license and copyright notice
b) state its changes to the code
c) disclose its source code as well
d) release the modified app under the same license 

In the majority of cases, copycats violate at least b), c), and d). Some even violate a) by proactively removing the notice that was there from the original project.
That is, by violating the terms and conditions of GPLv3, they illegally make use of the code.

For a prominent example of the long path to license compliance, see Tesla starting to open source parts of their software.

### Emotional/subjective reasons

In reality, many copycat apps on Google Play simply don’t add any value to the copied app. Most often they add some tracking libraries and ads to make money off it, but they rarely add any new features that improve the app for its users.

In the open source world, forking software is both very common and accepted.
For example KeepassXC, forked from KeepassX, is not at all frowned upon, because they adhere to the terms of the license and have added a ton of additional features that improved the app.

The copycat behaviour hurts the idea of open source, especially with respect to non-commercial open source projects. You as a developer have spend hours of your (often free) time to build something.
Often this is very rewarding: you start seeing people contributing to the project and caring about it. But in cases like this, there’s the feeling that the spirit of open source is being betrayed for egoistic reasons.
Yes, commercial use is allowed, but it would hurt much less if the company who made money of your work would not violate the terms of the license so blatantly.

In summary: problems arise when people fork/copy software for their own purposes, don’t comply with the license, and don’t actually improve the product.

**Comment by jaynecoding** [comment]
@thgoebel i respect your emotions, but as you know, the law is oblivious to emotions. this is so because different people will have differing emotions/opinions about the same subject. thats why laws are written in strict verbiage (_mostly_).
notwithstanding, i don't think it's healthy or proper use of time (as @tibbi alluded to) to hunt down people using this open source code and effectively shame them here; or force their removal from the play store. 
at the end of the day, this is open source code; and if author is worried about people using the code, he could license it under a different license scheme.

**Comment by tibbi** [comment]
@jaynecoding  Well, you apparently aren't working on open source projects. If I work on something for years, then someone just copies the project and adds some ads to it then republishes and makes a lot of money, it obviously sucks. The contributors at this project wouldn't like that either. And ye, as written previously, GNU GPL v3 requires the copycats to publish the source code, which they obviously don't and Google doesn't care about it.

**Comment by jaynecoding** [comment]
"If I work on something for years, then someone just copies the project and adds some ads to it then republishes and makes a lot of money, it obviously sucks."

there's a solution to that: don't open source your project.

**Comment by jaynecoding** [comment]
another very strange thing you did: when your code is ran, there is a pop up message that says "you're using a fake copy...." 

and there's direction to go buy your paid version.

using a fake copy? your open source code is fake??

it's as if you wanted the benefits of classifying your project as open source, but still want to enjoy the benefits of a closed system where users can only get it through one commercialized channel - via your paid app on the play store?
how do you reconcile this?

**Comment by michaelof** [comment]
> @jaynecoding Well, you apparently aren't working on open source projects. 

@tibbi you've already closed this issue, best solution. But just my 5 cents - better don‘t waste your time discussing with 2 days old fake accounts. Probably one of your copycats :)

**Comment by jaynecoding** [comment]
@michaelof it doesn't matter if the account is 2 seconds old. it doesn't matter whether i'm working on an open source project or not. and it doesn't matter if the account belongs to a "copy cat" (a term i find interesting, to say the least). in fact, you don't understand what open source is, if you call someone using an open source code as "copy cat". 
your comment wasn't worth 5 cents.

**Comment by Poussinou** [comment]
https://play.google.com/store/apps/details?id=kk.gallery

**Comment by tibbi** [comment]
thanks for the bad reviews :)
